### PR TITLE
feature: generate crossref relations from pub connections

### PR DESCRIPTION
**Pull request by 3mcd**
This PR improves the Crossref deposit submission process by automatically generating relations from Pub Connections. The relations that would be generated for a deposit can be reviewed in the `AssignDoiPreview` component in both human readable and XML formats. 

**Test Plan**
* Create a Pub with one or more inbound _and_ outbound connections.
* Click the "Assign DOI" (or "Resubmit DOI deposit") button.
* Preview the connections that would be deposited in the `AssignDoiPreview` component:

    Screen Shot 2020-07-28 at 2 53 01 PM
* Repeat for a Pub whose primary collection is a Conference and a Pub whose primary collection is a Book.

### CODE REVIEW

File: utils/crossref/transform/pub.js [hunk 1]
```diff
@@ -1,18 +1,71 @@
...
+		}
+	} else {
+		if (targetPub.doi) {
+			identifier = targetPub.doi;
+			identifierType = 'doi';
+		} else {
+			identifier = pubUrl(targetPub.community, targetPub);
+			identifierType = 'uri';
+		}
+	}
```
Comment (3mcd) [inline utils/crossref/transform/pub.js:30]: Kind of an ugly block of code - open to suggestions

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
 	},
 	"main": "init.js",
 	"scripts": {
-		"api-dev": "NODE_PATH=./client:./ node init.js --watch ./server --watch ./shared --watch ./client",
+		"api-dev": "NODE_PATH=./client:./ node init.js --watch ./server --watch ./utils --watch ./client",
```
Comment (idreyn) [inline package.json:13]: Aha! I knew something was up here!

File: server/doi/queries.js [hunk 3]
```diff
@@ -42,14 +41,12 @@ const findCollection = (collectionId) =>
...
 	Pub.findOne({
 		where: { id: pubId },
-		include: [
-			{ model: Release, as: 'releases' },
-			{
-				model: PubAttribution,
-				as: 'attributions',
-				include: [includeUserModel({ as: 'user' })],
+		...buildPubOptions({
+			getEdges: 'approved-only',
```
Comment (idreyn) [inline server/doi/queries.js:45]: This is the default and I think not strictly necessary.

**Comment by gabestein** [comment]
I won't be able to get to it today, but will take a look at the deposit xml on the deployed qubqub first thing tomorrow.

File: package.json (hunk 2 above)
Comment (3mcd) [reply package.json:13]: Yeah it was driving me crazy!

**Pull request merged**